api/src/test/java/org/apache/iceberg/TestIcebergBuild.java

```diff
@@ -37,6 +41,34 @@ public void testFullVersion() {
                 + ")");
   }
 
+  @Test
+  public void testVersionNotUnspecified() {
+    assertThat(IcebergBuild.version()).isNotEqualTo("unspecified");
+  }
+
+  @Test
+  public void testVersionMatchesSystemProperty() {
+    assumeThat(System.getProperty("project.version")).isNotNull();
+    assertThat(IcebergBuild.version())
+        .isEqualTo(System.getProperty("project.version"))
+        .as("IcebergBuild.version() should match system property project.version");
```

Comment:
".as(...)" needs to be used before the condition check of `isEqualTo`

Reply:
This is already merged, do you think it's worth changing in this cherrypick?

Reply:
no need to change this cherrypick. 

maybe we can just fix the main branch later.

Reply:
Thanks!